internal/stacks/metadata.go

```diff
@@ -0,0 +1,66 @@
+package stacks
+
+import (
+	"bytes"
+	"emperror.dev/errors"
+	"encoding/json"
+	"github.com/aviator-co/av/internal/git"
+	"github.com/sirupsen/logrus"
+)
+
+type BranchMetadata struct {
+	// The branch name associated with this stack.
+	// Not stored in JSON because the name can always be derived from the name
+	// of the git ref.
+	Name string `json:"-"`
+	// The branch name associated with the parent of the stack (if any).
+	Parent string `json:"parent"`
+}
+
+// GetMetadata loads information about the branch from the git repository.
+// Returns nil if given branch does not exist or is not a stacked branch.
+func GetMetadata(repo *git.Repo, branchName string) *BranchMetadata {
+	refName := stackMetadataRefName(branchName)
+	blob, err := repo.Git("cat-file", "blob", refName)
+
+	// Just assume that any error here means that the metadata ref doesn't exist
+	// (there's no easy way to distinguish between that and an actual Git error)
+	if err != nil {
+		return nil
+	}
+
+	var branch BranchMetadata
+	if err := json.Unmarshal([]byte(blob), &branch); err != nil {
+		logrus.WithError(err).WithField("ref", refName).Error("corrupt stack metadata, deleting...")
+		_ = repo.UpdateRef(&git.UpdateRef{Ref: refName, New: git.Missing})
+		return nil
+	}
+
+	return &branch
+}
+
+func writeMetadata(repo *git.Repo, s *BranchMetadata) error {
```

Comment:
shouldn't this be uppercase if called outside of `metadata.go`

Reply:
No, it's private within the package, not the file.